sumologic/resource_sumologic_data_forwarding_test.go

```diff
@@ -0,0 +1,201 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDataForwardingExists(),
+					resource.TestCheckResourceAttr(dataForwardingResourceName, "destination_name", destinationName),
+					resource.TestCheckResourceAttr(dataForwardingResourceName, "description", description),
+					resource.TestCheckResourceAttr(dataForwardingResourceName, "s3_region", region),
+					resource.TestCheckResourceAttr(dataForwardingResourceName, "s3_server_side_encryption", "false"),
+				),
+			}, {
+				Config: testAccSumologicDataForwardingUpdateConfig(destinationName, description, testAwsBucket, testAwsRoleArn, region),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDataForwardingExists(),
+					resource.TestCheckResourceAttr(dataForwardingResourceName, "destination_name", destinationName),
+					resource.TestCheckResourceAttr(dataForwardingResourceName, "description", description),
+					resource.TestCheckResourceAttr(dataForwardingResourceName, "s3_region", region),
+					resource.TestCheckResourceAttr(dataForwardingResourceName, "s3_server_side_encryption", "true"),
+				),
+			},
+		},
+	})
+
+}
+
+func TestAccSumologicDataForwarding_delete(t *testing.T) {
+	dataForwardingResourceName, destinationName, description, region, testAwsRoleArn, testAwsBucket := getTestParams()
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheckWithAWS(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSumologicDataForwardingCreateConfig(destinationName, description, testAwsBucket, testAwsRoleArn, region),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDataForwardingExists(),
+					resource.TestCheckResourceAttr(dataForwardingResourceName, "destination_name", destinationName),
+					resource.TestCheckResourceAttr(dataForwardingResourceName, "description", description),
+					resource.TestCheckResourceAttr(dataForwardingResourceName, "s3_region", region),
+					resource.TestCheckResourceAttr(dataForwardingResourceName, "s3_server_side_encryption", "false"),
+				),
+			}, {
+				Config: testAccSumologicDataForwardingDeleteConfig(),
+				Check:  resource.ComposeTestCheckFunc(testAccCheckDataForwardingDestroy()),
+			},
+		},
+	})
+}
+
+func testAccCheckDataForwardingExists() resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := testAccProvider.Meta().(*Client)
+		for _, r := range s.RootModule().Resources {
+			id := r.Primary.ID
+			if _, err := client.getDataForwarding(id); err != nil {
+				return fmt.Errorf("Received an error retrieving data forwarding %s", err)
+			}
+		}
+		return nil
+	}
+}
+
+func testAccCheckDataForwardingDestroy() resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := testAccProvider.Meta().(*Client)
+		for _, r := range s.RootModule().Resources {
+			id := r.Primary.ID
+			p, err := client.getDataForwarding(id)
+
+			if err != nil {
```

Comment:
Is this a good practice? The reason why I did it is to check if the resource is no longer there. Doing a get on the id of the resource that does not exist throws an error. So if I am getting that particular error that mean we are good and the resource is gone.